src/litserve/specs/openai_embedding.py

```diff
@@ -56,36 +55,35 @@ class EmbeddingResponse(BaseModel):
     usage: UsageInfo
 
 
-LITAPI_VALIDATION_MSG = """LitAPI.predict and LitAPI.encode_response must not be a generator\
-(use return instead of yield) while using the OpenAI embedding spec.
-
-Error: {}
-
-Please follow the below example for guidance on how to use the spec:
+EMBEDDING_API_EXAMPLE = """
+Please follow the example below for guidance on how to use the OpenAI Embedding spec:
 
 ```python
 import numpy as np
 from typing import List
-from litserve.api import LitAPI
+from litserve import LitAPI, OpenAIEmbeddingSpec
 
 
 class TestAPI(LitAPI):
     def setup(self, device):
         self.model = None
 
-    def decode_request(self, request)->List[str]:
+    def decode_request(self, request) -> List[str]:
         return request.input if isinstance(request.input, list) else [request.input]
 
-    def predict(self, x)-> List[List[float]]:
+    def predict(self, x) -> List[List[float]]:
         return np.random.rand(len(x), 768).tolist()
 
-    def encode_response(self, output)-> dict:
+    def encode_response(self, output) -> dict:
         return {"embeddings": output}
```

Comment:
why do we need a list of strings to be returned from `decode_request`? Probably for simplicity of example, predict can do something like `np.random.rand(len([x] if isinstance(x, str) else x, 768).tolist()`

Reply:
or use `request.ensure_list`

Reply:
Good point, @aniketmaurya. That approach definitely simplifies things and keeps the logic flexible.

Reply:
So , basically we can leave that up to user how they want to return from decode request: i.e either string to list of strings